src/power_grid_model/core/serialization.py

```diff
@@ -90,16 +90,11 @@ class Serializer(ABC):
     Serializer for the Power grid model
     """
 
-    _data: Mapping[ComponentType, np.ndarray] | Mapping[ComponentType, np.ndarray | Mapping[str, np.ndarray]]
+    _data: Dataset
     _dataset: CConstDataset
     _serializer: SerializerPtr
 
-    def __new__(
-        cls,
-        data: Mapping[ComponentType, np.ndarray] | Mapping[ComponentType, np.ndarray | Mapping[str, np.ndarray]],
-        serialization_type: SerializationType,
-        dataset_type: Optional[DatasetType] = None,
-    ):
+    def __new__(cls, data: Dataset, serialization_type: SerializationType, dataset_type: Optional[DatasetType] = None):
```

Comment:
By making this change, it would allow column data too right? Shouldnt this be handled in python step 3 4 PR?

Reply:
this is why i did not uncomment the data types linking columnar data to the dataset yet (together with TODO statements)